tools/VscPreviewExtension/src/extension.ts

```diff
@@ -64,35 +77,28 @@ export function activate(context: ExtensionContext) {
     let server = http.createServer();
     server.on("request", function (req, res) {
         let requestInfo = req.url.split("/");
-        if (requestInfo[1] === ConstVariable.matchFromR2L) {
-            if (!mapToSelection(parseInt(requestInfo[2]), parseInt(requestInfo[3])))
-                window.showErrorMessage("Selection Range Error");
-        } else if (requestInfo[1] === ConstVariable.matchFromL2R) {
-            res.writeHead(200, { "Content-Type": "text/plain" });
-            res.write(startLine + " " + endLine);
-            res.end();
+        switch (requestInfo[1]) {
+            case ConstVariable.matchFromR2L:
+                if (!mapToSelection(parseInt(requestInfo[2]), parseInt(requestInfo[3])))
+                    window.showErrorMessage("Selection Range Error");
+                res.end();
+                break;
+            case ConstVariable.matchFromL2R:
+                res.writeHead(200, { "Content-Type": "text/plain" });
+                res.write(startLine + " " + endLine);
+                res.end();
         }
     });
-    let port = getRandomInt(50000, 65535);
-    let isPortOccupied = false;
-    while (true) {
-        server.listen(port)
-            .on("error", function (error) {
-                port = getRandomInt(50000, 65535);
-                isPortOccupied = true;
-            });
-        if (!isPortOccupied) {
-            dfmProcess.provider.port = port;
-            tokenTreeProcess.provider.port = port;
-            break;
-        }
-    }
+
+    server.listen(0);
```

Comment:
No, this server is for two side match.

---
In reply to: [111559151](https://github.com/dotnet/docfx/pull/1547#discussion_r111559151) [](ancestors = 111559151)